x-pack/plugins/snapshot_restore/plugin.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Legacy } from 'kibana';
+import { PLUGIN } from './common/constants';
+import { registerRoutes } from './server/routes/api/register_routes';
+import { createShim } from './shim';
+
+export class Plugin {
+  public server: Legacy.Server;
+
+  constructor(server: Legacy.Server) {
+    this.server = server;
+  }
+
+  public start(): void {
+    const server = this.server;
+    const { core, plugins } = createShim(server, PLUGIN.ID);
```

Comment:
I recommend pulling the shimming logic into `snapshot_restore/index.ts`, so `plugin.ts` doesn't appear coupled to the legacy stuff at all. So you'd call `createShim` from the legacy `init()` and pass `core` and `plugins` as arguments to this `start()`.

Over time you'll convert more of the shim stuff over to new platform services directly, and then eventually you'll just delete the legacy plugin definition entirely, and your new plugin definition won't really change along the way.